env0/data_source_code_variables.go

```diff
@@ -0,0 +1,81 @@
+package env0
+
+import (
+	"context"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceCodeVariables() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCodeVariablesRead,
+
+		Schema: map[string]*schema.Schema{
+			"template_id": {
+				Type:        schema.TypeString,
+				Description: "extracts source code terraform variables from the VCS configuration of this template",
+				Required:    true,
+			},
+
+			"variables": {
+				Type:        schema.TypeList,
+				Description: "a list of terraform variables extracted from the source code",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "the name of the terraform variable",
+							Computed:    true,
+						},
+						"value": {
+							Type:        schema.TypeString,
```

Comment:
I think that there are some missing properties. Regardless `schema_enum`(which I don't think is required) there are `is_sensitive`, `description`